task-native/src/main/java/org/ballerinalang/stdlib/task/actions/TaskActions.java

```diff
@@ -42,94 +38,93 @@
 public class TaskActions {
 
     public static Object pause(BObject taskListener) {
-        Task task = (Task) taskListener.getNativeData(NATIVE_DATA_TASK_OBJECT);
+        String triggerID = (String) taskListener.getNativeData(TaskConstants.TRIGGER_NAME);
+        TaskScheduler taskScheduler = (TaskScheduler) taskListener.getNativeData(TaskConstants.SCHEDULER);
         try {
-            task.pause();
-        } catch (SchedulingException e) {
-            return createTaskError(TaskConstants.SCHEDULER_ERROR, e.getMessage());
+            taskScheduler.pause(triggerID);
+        } catch (SchedulerException e) {
+            return Utils.createTaskError("Cannot pause Listener/Scheduler." + e.getMessage());
         }
         return null;
     }
 
     public static Object resume(BObject taskListener) {
-        Task task = (Task) taskListener.getNativeData(NATIVE_DATA_TASK_OBJECT);
+        String triggerID = (String) taskListener.getNativeData(TaskConstants.TRIGGER_NAME);
+        TaskScheduler taskScheduler = (TaskScheduler) taskListener.getNativeData(TaskConstants.SCHEDULER);
         try {
-            task.resume();
-        } catch (SchedulingException e) {
-            return createTaskError(TaskConstants.SCHEDULER_ERROR, e.getMessage());
+            taskScheduler.resume(triggerID);
+        } catch (SchedulerException e) {
+            return Utils.createTaskError("Cannot resume Listener/Scheduler." + e.getMessage());
```

Comment:
```suggestion
            return Utils.createTaskError("Cannot resume the Listener/Scheduler. " + e.getMessage());
```

Check for the other places as well.